src/widgets.rs

```diff
@@ -561,4 +606,42 @@ mod tests {
         widget.render_ref(buf.area, &mut buf);
         assert_eq!(buf, Buffer::with_lines(["                    "]));
     }
+
+    #[rstest]
+    fn str_render(mut buf: Buffer) {
+        let widget = "hello world";
+        widget.render_ref(buf.area, &mut buf);
+        assert_eq!(buf, Buffer::with_lines(["hello world         "]));
+
+        let widget = "hello world";
+        widget.render(buf.area, &mut buf);
+        assert_eq!(buf, Buffer::with_lines(["hello world         "]));
```

Comment:
Rendering to the same buffer instead of a new one is a bad way to test as the first rendering already updates the buffer with the expected output. Split these into 4 tests or reconstruct the buffer on each branch of the test. That might be easier to read as just the following. It's probably also worth writing this as `"hello world".render_ref(...)` to be canonical with the intended usage.

```rust
let buf = &mut single_line_buffer();
"hello world!".render(buf.area, buf);
assert_eq!(buf, Buffer::with_lines(["hello world!        "]), "str.render()");

let buf = &mut single_line_buffer();
String::new("hello world!").render(buf.area, buf);
assert_eq!(buf, Buffer::with_lines(["hello world!        "]), "String.render()");
```